src/algorithms/tracking/ActsToTracks.cc

```diff
@@ -185,6 +207,22 @@ void ActsToTracks::process(const Input& input, const Output& output) const {
 
       });
 
+      // Store track associations if hit associations provided
+      // FIXME: not able to check whether optional inputs were provided
+      //if (raw_hit_assocs->has_value()) {
+        double total_weight = std::accumulate(
+          mcparticle_weight_by_hit_count.begin(), mcparticle_weight_by_hit_count.end(),
+          0, [](const double sum, const auto& i) { return sum + i.second; });
+        for (const auto& [mcparticle, weight] : mcparticle_weight_by_hit_count) {
```

Comment:
So, if a given track has measurement hits coming from multiple MCParticles, you create an association with each 
of these MCParticle; not just the MCParticle with the largest weight. This saves more information, which is good, but we'll need to  inform users that this is what the association contains.

Reply:
This is how all associations are done.